lib/cudnn/src/cuDNN.jl

```diff
@@ -163,9 +155,21 @@ function __init__()
 
     CUDA.functional() || return
 
-    if !CUDNN_jll.is_available()
-        @error "cuDNN is not available for your platform ($(Base.BinaryPlatforms.triplet(CUDNN_jll.host_platform)))"
-        return
+    global libcudnn
```

Comment:
That's a lot of code reuse with just a lib name change no? Could it be maybe wrapped in a generic "init" to reuse with lib name and jll name?

Reply:
Only 3 copies doesn't seem worth an abstraction. Furthermore, it will diverge when we add some better error reporting (e.g. CUTENSOR not being available on CUDA 12.1 probably should prompt users to do `set_runtime_version!(v"11.8")`).

Reply:
Sounds good